consensus/replay_test.go

```diff
@@ -104,3 +150,68 @@ func openWAL(t *testing.T, cs *ConsensusState, file string) {
 	wal.exists = true
 	cs.wal = wal
 }
+
+//-----------------------------------------------
+// Test the log at every iteration, and set the privVal last step
+// as if the log was written after signing, before the crash
+
+func TestReplayCrashAfterWrite(t *testing.T) {
+	split := strings.Split(testLog, "\n")
+	for i := 0; i < len(split)-1; i++ {
+		cs, newBlockCh, _, f := setupReplayTest(i+1, true)
+		runReplayTest(t, cs, f, newBlockCh)
```

Comment:
the block we wait for is just a signal that we haven't halted, its not necessarily something contained in the WAL itself. Assuming the consensus state is running, replay of any WAL, including the empty one, should eventually be followed by a new block, or else something is wrong